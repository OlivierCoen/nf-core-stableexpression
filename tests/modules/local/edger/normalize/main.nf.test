nextflow_process {

    name "Test Process EDGER_NORMALIZE"
    script "modules/local/edger/normalize/main.nf"
    process "EDGER_NORMALIZE"
    tag "edger_normalize"
    tag "module"

    test("Should run without failures") {

        when {

            process {
                """
                meta = [accession: "accession", design: file('$baseDir/tests/input/normalize/design.csv')]
                input[0] = [meta, file('$baseDir/tests/input/normalize/all_counts.csv')]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
