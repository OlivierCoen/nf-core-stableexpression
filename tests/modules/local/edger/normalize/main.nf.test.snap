{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "accession": "accession",
                            "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                        },
                        "all_counts.log_cpm.csv:md5,9dd697940642a7e2dbac520eb7a2eeff"
                    ]
                ],
                "1": [
                    [
                        "EDGER_NORMALIZE",
                        "R",
                        "4.3.3 (2024-02-29)"
                    ]
                ],
                "2": [
                    [
                        "EDGER_NORMALIZE",
                        "edgeR",
                        "4.0.16"
                    ]
                ],
                "csv": [
                    [
                        {
                            "accession": "accession",
                            "design": "design.csv:md5,a83dd6a15463b51d94f0a42c196d7933"
                        },
                        "all_counts.log_cpm.csv:md5,9dd697940642a7e2dbac520eb7a2eeff"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-29T12:36:28.080031"
    }
}